### PR TITLE
fix: Locale overrides should only allow one row per machine

### PR DESCRIPTION
**Pull request by liordino**
fix: filter already overriden machines on locales

### CODE REVIEW

**Comment by liordino** [comment]
[Locale overrides should only allow one row per machine](https://trello.com/c/5IvfE8nD/190-locale-overrides-should-only-allow-one-row-per-machine)

File: new-lamassu-admin/src/pages/Locales/Locales.js [hunk 1]
```diff
@@ -69,6 +69,18 @@ const Locales = ({ name: SCREEN_KEY }) => {
...
+      const overridenMachines = locale.overrides?.map(o => o.machine)
+      const machinesIndex = data.findIndex(o => o.name === 'machine')
+      data[machinesIndex].inputProps.options = data[
+        machinesIndex
+      ].inputProps.options?.filter(m => !overridenMachines.includes(m.deviceId))
+    }
+
+    return data
+  }
+
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Locales/Locales.js]: This wont work. 

- empty overrides table break the page.
- already existing overrides won't show the machine when being edited.

Take a look at how the FiatBalanceOverrides does it. Better to just follow that logic.

File: new-lamassu-admin/src/pages/Locales/helper.js [hunk 2]
```diff
@@ -39,7 +54,11 @@ const allFields = getData => {
       view: getView(machineData, 'name', 'deviceId'),
       input: Autocomplete,
       inputProps: {
-        options: machineData,
+        options: it =>
+          R.concat(
+            suggestionFilter(machineData),
+            findSuggestion(it, machineData)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Locales/helper.js]: Probably better to do

```
R.concat(it?.machine ? [it.machine] : [])(suggestionFilter...)
```

After the changes on Autocomplete the code checks for id to match a option, so I don't think we need to fetch it within the machineData.

File: new-lamassu-admin/src/pages/Locales/helper.js [hunk 3]
```diff
@@ -20,12 +23,24 @@ const allFields = getData => {
...
   const displayCodeArray = data => it => {
     if (!it) return it
 
     return R.compose(R.join(', '), R.map(getView(data, 'code')))(it)
   }
 
+  const overridenMachines = R.map(override => override.machine, auxElements)
+  const suggestionFilter = R.filter(
+    it => !R.contains(it.deviceId, overridenMachines)
+  )
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Locales/helper.js]: Better to use https://ramdajs.com/docs/#differenceWith

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**